### ISSUE TITLE
False positive display-name when using forwardRef

### ISSUE THREAD

**Issue by Sharcoux** [eslint/eslint#11711]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.12.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint 10.0.1

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended"
    ],
    "parserOptions": {
        "ecmaVersion": 9,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "indent": [
            "error",
            2,
            { "SwitchCase": 1 }
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "no-console": "warn",
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import PropTypes from 'prop-types';
import React from 'react';

const Paragraph = React.forwardRef(function Paragraph({ style }, ref) {
  // const finalStyle = Object.assign({}, style);
  return <div ref={ref} style={style} />;
});

Paragraph.propTypes = {
  style: PropTypes.object
};

export default Paragraph;
export const proptype = Paragraph.PropTypes;
```

```bash
npx eslint <file>
```

**What did you expect to happen?**
Nothing as the file is valid

**What actually happened? Please include the actual, raw output from ESLint.**

```
  4:19  error  Component definition is missing display name  react/display-name

✖ 1 problem (1 error, 0 warnings)
```

**Fun fact!** If you uncomment the line with Object.assign, the false positive disappears...

**Are you willing to submit a pull request to fix this bug?**

**Comment by platinumazure** [comment]
Hi @Sharcoux, thanks for the issue.

This is an issue with eslint-plugin-react. Could you please [file an issue](https://github.com/yannickr/eslint-plugin-react) there? Thanks!

**Comment by Sharcoux** [comment]
Oh, ok. Thanks.

**Comment by kaicataldo** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
